packages/data/src/onboarding/selectors.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * Internal dependencies
+ */
+import { WPDataSelectors } from '../types';
+
+export const getProfileItems = (
+	state: OnboardingState
+): ProfileItemsState | Record< string, never > => {
+	return state.profileItems || {};
+};
+
+export const getTasksStatus = (
+	state: OnboardingState
+): TasksStatusState | Record< string, never > => {
+	return state.tasksStatus || {};
+};
+
+export const getOnboardingError = (
+	state: OnboardingState,
+	selector: string
+): unknown | false => {
+	return state.errors[ selector ] || false;
+};
+
+export const isOnboardingRequesting = (
+	state: OnboardingState,
+	selector: string
+): boolean => {
+	return state.requesting[ selector ] || false;
+};
+
+// Types
+export type OnboardingSelectors = {
+	getProfileItems: () => ReturnType< typeof getProfileItems >;
+	getTasksStatus: () => ReturnType< typeof getTasksStatus >;
+	getOnboardingError: () => ReturnType< typeof getOnboardingError >;
+	isOnboardingRequesting: () => ReturnType< typeof isOnboardingRequesting >;
+} & WPDataSelectors;
+
+export type OnboardingState = {
+	profileItems: ProfileItemsState;
+	tasksStatus: TasksStatusState;
+	// TODO clarify what the error record's type is
+	errors: Record< string, unknown >;
+	requesting: Record< string, boolean >;
+};
+
+export type TasksStatusState = {
+	automatedTaxSupportedCountries: string[];
+	enabledPaymentGateways: string[];
+	hasHomepage: boolean;
+	hasPaymentGateway: boolean;
+	hasPhysicalProducts: boolean;
+	hasProducts: boolean;
+	isAppearanceComplete: boolean;
+	isTaxComplete: boolean;
+	shippingZonesCount: number;
+	stripeSupportedCountries: string[];
+	stylesheet: string;
+	taxJarActivated: boolean;
+	// TODO - fill out this type
+	themeMods: unknown;
+	wcPayIsConnected: false;
+};
+
+export type Industry = {
+	slug: string;
+};
+
+export type ProductCount = '0' | '1-10' | '11-100' | '101 - 1000' | '1000+';
+
+export type ProductTypeSlug =
+	| 'physical'
+	| 'bookings'
+	| 'download'
+	| 'memberships'
+	| 'product-add-ons'
+	| 'product-bundles'
+	| 'subscriptions';
+
+export type OtherPlatformSlug =
+	| 'shopify'
+	| 'bigcommerce'
+	| 'wix'
+	| 'amazon'
+	| 'ebay'
+	| 'etsy'
+	| 'squarespace'
+	| 'other';
```

Comment:
This is handy, although it will be an extra item to keep track off when updating the onboarding flow.

Reply:
@louwie17 there will always be manual updating to do with these kinds of types for sure, but I think its mostly helpful to have it self-documented and keep track of possible values.